app.py

```diff
@@ -83,27 +81,95 @@ def get_db_collection():
 def create_app():
     app = Flask(__name__)
     app.config.from_object('flask_config.Config')
+    login_disabled = os.getenv('LOGIN_DISABLED').upper() == 'TRUE'
+    app.config['LOGIN_DISABLED'] = login_disabled
 
     collection = get_db_collection()
 
+    login_manager = LoginManager()
+
+    client_id = os.getenv('CLIENT_ID')
+    client_secret = os.getenv('CLIENT_SECRET')
+    state = ''.join(random.choice(string.ascii_lowercase) for i in range(8))
```

Comment:
You're creating a single state parameter per application instance; that means:
* If a user hits two different instances of our server, they'll fail the authentication
* If our application has a long lifetime, then other services will still be able to find & take our state & use it for CSRF

If we're using state, we should instead:
* Generate it fresh for each authentication attempt
* Store it on the client's device (e.g. in their session), so that we can verify when they come back that the state Github received matches the one we gave that client